core/services/vrf/listener_v2.go

```diff
@@ -460,17 +460,24 @@ func (lsn *listenerV2) shouldProcessSub(subID uint64, sub vrf_coordinator_v2.Get
 
 	gasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddress)
 
-	estimatedFee, err := lsn.estimateFeeJuels(vrfRequest, gasPriceWei)
-	if err != nil {
-		l.Warnw("Couldn't estimate fee, processing sub anyway", "err", err)
-		return true
-	}
+	simResult := lsn.simulateFulfillment(ctx, gasPriceWei, reqs[0], l)
+	if simResult.err != nil {
+		// if the simulation fails with "execution reverted" it is most likely insufficient funds.
+		// in that case don't process the subscription.
+		if sub.Balance.Cmp(simResult.juelsNeeded) < 0 && errors.Is(simResult.err, errPossiblyInsufficientFunds{}) {
+			l.Infow(
+				"Insufficient link balance to fulfill a request based on estimate and simulation, not processing sub",
+				"err", simResult.err,
+				"juelsNeeded", simResult.juelsNeeded,
+			)
+			return false
+		}
 
-	if sub.Balance.Cmp(estimatedFee) < 0 {
-		l.Infow("Subscription is underfunded, not processing it's requests",
-			"estimatedFeeJuels", estimatedFee,
-		)
-		return false
+		// in the event simulation fails due to a missing blockhash or some other reason,
+		// we still process the subscription in case the blockhash will become available at
+		// some future point in time.
+		l.Warnw("Pipeline run failed, processing sub anyway", "err", simResult.err)
+		return true
```

Comment:
This seems like the safest approach, since just because one request failed to e.g have a blockhash it doesn't mean all requests are in that bad of shape.